x/config/config.go

```diff
@@ -171,3 +171,27 @@ func NewPacketListener(transportConfig string) (transport.PacketListener, error)
 		return nil, fmt.Errorf("config scheme '%v' is not supported", url.Scheme)
 	}
 }
+
+func SanitizeConfig(transportConfig string) (string, error) {
+	// Do nothing if the config is empty
+	if transportConfig == "" {
+		return "", nil
+	}
+	// Split the string into parts
+	parts := strings.Split(transportConfig, "|")
+
+	// Iterate through each part
+	for i, part := range parts {
+		u, err := parseConfigPart(part)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse config part: %w", err)
+		}
+		if u.User != nil {
+			u.User = url.User("redacted")
```

Comment:
Perhaps something that makes it clearer it's a placeholder.
```suggestion
			u.User = url.User("[REDACTED]")
```